pkg/kubeapiserver/options/authentication.go

```diff
@@ -397,16 +400,51 @@ func (o *BuiltInAuthenticationOptions) ToAuthenticationConfig() (kubeauthenticat
 		}
 	}
 
-	if o.OIDC != nil {
-		ret.OIDCCAFile = o.OIDC.CAFile
-		ret.OIDCClientID = o.OIDC.ClientID
-		ret.OIDCGroupsClaim = o.OIDC.GroupsClaim
-		ret.OIDCGroupsPrefix = o.OIDC.GroupsPrefix
-		ret.OIDCIssuerURL = o.OIDC.IssuerURL
-		ret.OIDCUsernameClaim = o.OIDC.UsernameClaim
-		ret.OIDCUsernamePrefix = o.OIDC.UsernamePrefix
+	if o.OIDC != nil && len(o.OIDC.IssuerURL) > 0 && len(o.OIDC.ClientID) > 0 {
+		jwtAuthenticator := authenticationapi.JWTAuthenticator{
+			Issuer: authenticationapi.Issuer{
+				URL:       o.OIDC.IssuerURL,
+				ClientIDs: []string{o.OIDC.ClientID},
+			},
+			ClaimMappings: authenticationapi.ClaimMappings{
+				Groups: authenticationapi.PrefixedClaimOrExpression{
+					Prefix: o.OIDC.GroupsPrefix,
+					Claim:  o.OIDC.GroupsClaim,
+				},
+				Username: authenticationapi.PrefixedClaimOrExpression{
+					Prefix: o.OIDC.UsernamePrefix,
+					Claim:  o.OIDC.UsernameClaim,
+				},
+			},
+		}
+
+		if len(o.OIDC.CAFile) != 0 {
+			var err error
+			jwtAuthenticator.Issuer.CertificateAuthority, err = os.ReadFile(o.OIDC.CAFile)
+			if err != nil {
+				return kubeauthenticator.Config{}, err
+			}
+		}
+
+		if len(o.OIDC.RequiredClaims) > 0 {
+			claimValidationRules := make([]authenticationapi.ClaimValidationRule, 0, len(o.OIDC.RequiredClaims))
+			for claim, value := range o.OIDC.RequiredClaims {
+				claimValidationRules = append(claimValidationRules, authenticationapi.ClaimValidationRule{
+					Claim:         claim,
+					RequiredValue: value,
+				})
+			}
+			jwtAuthenticator.ClaimValidationRules = claimValidationRules
+		}
+
+		authConfig := &authenticationapi.AuthenticationConfiguration{
+			JWT: []authenticationapi.JWTAuthenticator{jwtAuthenticator},
+		}
+		if fieldErrList := authenticationapivalidation.ValidateAuthenticationConfiguration(authConfig); fieldErrList.ToAggregate() != nil {
```

Comment:
I am not sure if this is the correct place to call the validation logic, normally options are validate right after parsing in a `Validate` function.

Reply:
As discussed last week, I've left the validation here without any changes. LMK what you think.